### ISSUE TITLE
<mj-button> cannot take an rgb background-color

### ISSUE THREAD

**Issue by john-halbert** [mjmlio/mjml#605]
The `mj-button` accepts a `color` unit in the `background-color` parameter, which seems to be only hinted at in the [banner](https://mjml.io/documentation/#banner) documentation as:
> Note here that colors can be expressed using different formats (rgb, hsl, hex etc.).

It looks like the `mj-button` tag creates a `td` with a `bgcolor` attribute. It appears that [browser-based clients won't render this](http://stackoverflow.com/a/10201434), resulting in the `p` only being colored correctly, not the entire button. 

**Possible solutions:** 
- Add a redundant `background-color` rule in the `td` that only has a bgcolor property. 
- Convert values to hex at compile-time?
- The documentation could be updated to warn about this edge case, or explicitly state hex codes (they are there by example, but no such case is mentioned.)

Here's the code to reproduce:

``` mjml
<mjml>
  <mj-body>
    <mj-container>
      <mj-section>
        <mj-button background-color="rgb(0,0,0)">
          hello world
        </mj-button>
      </mj-section>
    </mj-container>
  </mj-body>
</mjml>
```

Results in:

``` html
...
<table role="presentation" cellpadding="0" cellspacing="0" style="border-collapse:separate;" align="center" border="0">
    <tbody>
        <tr>
            <td style="border:none;border-radius:3px;color:#ffffff;cursor:auto;padding:10px 25px;" align="center" valign="middle" bgcolor="rgb(0,0,0)">
                <p style="text-decoration:none;line-height:100%;background:rgb(0,0,0);color:#ffffff;font-family:Ubuntu, Helvetica, Arial, sans-serif;font-size:13px;font-weight:normal;text-transform:none;margin:0px;">hello world</p>
            </td>
       </tr>
   </tbody>
</table>
...
```

Here's my suggested change:

```html
...
<table role="presentation" cellpadding="0" cellspacing="0" style="border-collapse:separate;" align="center" border="0">
    <tbody>
        <tr>
            <td style="background:rgb(0,0,0);border:none;border-radius:3px;color:#ffffff;cursor:auto;padding:10px 25px;" align="center" valign="middle" bgcolor="rgb(0,0,0)">
                <p style="text-decoration:none;line-height:100%;background:rgb(0,0,0);color:#ffffff;font-family:Ubuntu, Helvetica, Arial, sans-serif;font-size:13px;font-weight:normal;text-transform:none;margin:0px;">hello world</p>
            </td>
       </tr>
   </tbody>
</table>
...
```

Thanks for your time!

**Comment by ngarnier** [comment]
Hello @john-halbert, 

Thanks a lot for taking the time of investigating and reporting this issue! You're absolutely right, however this should be fix in 3.3.2 as we make the button clickable thanks to #557.

**Comment by jpsfs** [comment]
Hi,

I can confirm this is still happening now on the online editor.

Best,
José

**Comment by iRyusa** [comment]
As mentioned in the issue this will be fixed in mjml 4.2.

**Comment by kmcb777** [comment]
Hi @john-halbert @jpsfs this has been fixed in the new release mjml4.2-beta, which you can install with npm install mjml, you can check if it solves the problem
